### PR TITLE
feat: remove option for legacy hson files

### PR DESCRIPTION
**Pull request by AngRodrigues**
- Prevents the use of a config file as json or hjson. 
- Included an utils function to convert from old legacy file - `update_from_legacy_file`
- if the argument `config_filename` is used, returns ValueError. 

Usage example of converter function, using in the hjson file in [here](https://github.com/Loop3D/map2loop-notebooks/blob/master/source_data/example.hjson):

``` 
from map2loop.project import Project
from map2loop.utils import update_from_legacy_file


bounding_box = {
    "minx": 520000,  "miny": 7490000,  "maxx": 550000,  "maxy": 7510000,   "base": -3200,   "top": 1200,
}

loop_project_filename = "local_source.loop3d"

proj = Project( 
    geology_filename = r"source_data\geol_clip_no_gaps.shp", # required
    structure_filename = r"\source_data\structure_clip.shp", #required
    dtm_filename = r"source_data\dtm_rp.tif", # required
    config_dictionary = update_from_legacy_file(filename=r"source_data\example.hjson"),
    working_projection = "EPSG:28350",
    bounding_box = bounding_box,
    loop_project_filename = loop_project_filename,
    overwrite_loopprojectfile = True, 
)

### CODE REVIEW

File: map2loop/project.py [hunk 1]
```diff
@@ -204,18 +207,12 @@ def __init__(
         if fault_orientation_filename != "":
             self.map_data.set_filename(Datatype.FAULT_ORIENTATION, fault_orientation_filename)
 
-        if config_filename != "":
```
Comment (lachlangrose) [inline map2loop/project.py:207]: Do we want to remove the capability to load a config file from json? I would probably prefer to allow the config dictionary to be a path/string and be loaded using json.

File: map2loop/project.py [hunk 2]
```diff
@@ -147,6 +145,11 @@ def __init__(
         self.fold_samples = pandas.DataFrame(columns=["ID", "X", "Y", "Z", "featureId"])
         self.geology_samples = pandas.DataFrame(columns=["ID", "X", "Y", "Z", "featureId"])
 
+        # check if user is using a config file or dictionary, if file, break the project.
```
Comment (lachlangrose) [inline map2loop/project.py]: What if the file is a new config file not a legacy file?

File: map2loop/mapdata.py [hunk 3]
```diff
@@ -267,7 +267,7 @@ def set_config_filename(
                 Whether the file is in m2lv2 form. Defaults to False.
         """
         logger.info('Setting config filename to {filename}')
-        self.config_filename = filename
+
```
Comment (lachlangrose) [inline map2loop/mapdata.py]: Should the legacy_format argument be removed?

File: map2loop/project.py [hunk 4]
```diff
@@ -106,10 +106,8 @@ def __init__(
                 The filename of the configuration json file to use (if not using config_dictionary). Defaults to "".
             config_dictionary (dict, optional):
                 A dictionary version of the configuration file. Defaults to {}.
-            clut_filename (str, optional):
+            clut_filename (str, deprecated):
```
Comment (lachlangrose) [inline map2loop/project.py]: WHy is this deprecated? clut is the colours?

**Comment by AngRodrigues** [comment]
A note that I will merge this into 3.2 by COB today if no review

**Pull request merged**